lib/win32/taskscheduler/time_calc_helper.rb

```diff
@@ -128,6 +129,9 @@ def extra_days(days_count, month_count, year_count, init_month, init_year)
       def time_details(time_str)
         tm_detail = {}
         if time_str.to_s != ""
+          # raising exception if time is not a string
+          raise TypeError.new("TypeError: Inccorrect Type please pass input value as a String.") unless time_str.is_a? String
```

Comment:
This is great, but lets do it before the `if`

Reply:
tagging @antima-gupta  ~ please review the comments above.